apps/src/levelbuilder/unit-editor/UnitEditor.jsx

```diff
@@ -242,60 +229,16 @@ class UnitEditor extends React.Component {
         error: 'Please set both version year and family name.',
       });
       return;
-    } else if (
-      [PublishedState.preview, PublishedState.stable].includes(
-        this.state.publishedState
-      ) &&
-      this.props.isMissingRequiredDeviceCompatibilities
-    ) {
-      this.setState({
-        isSaving: false,
-        error:
-          'Please set all device compatibilities in order to save with published state as preview or stable.',
-      });
-      return;
-    }
-
-    if (this.state.publishedState !== this.props.initialPublishedState) {
-      const msg =
-        'It looks like you are updating the published state. ' +
-        'Are you sure you want to update the published state? ' +
-        'Once you update the published state you can not go back to this published state. ' +
-        'For example once you set the published state to beta you can not go back to in development. ' +
-        'Also once a course as a published state of pilot it can not be fully launched (marked as preview or stable).';
-      if (!window.confirm(msg)) {
-        this.setState({
-          isSaving: false,
-          error: 'Saving cancelled.',
-        });
-        return;
-      }
-    }
-
-    if (
-      this.state.unitPublishedState !== this.props.initialUnitPublishedState
-    ) {
-      const msg =
-        'It looks like you are hiding this unit. ' +
-        'Are you sure you want to hide this unit? ';
-      if (!window.confirm(msg)) {
-        this.setState({
-          isSaving: false,
-          error: 'Saving cancelled.',
-        });
-        return;
-      }
     }
 
     if (
-      this.state.unitPublishedState === PublishedState.in_development &&
-      this.state.unitPublishedState === this.props.initialUnitPublishedState
+      this.state.hideWithinCourse &&
+      this.state.hideWithinCourse !== this.props.initialHideWithinCourse
     ) {
       const msg =
-        'This unit is hidden within the course, meaning it is not ' +
-        'visible on the Course Overview page, Section Dialog, or Teacher ' +
-        'Dashboard. It is still visible to Levelbuilders. Would you ' +
-        'like to continue with saving?';
+        'Hiding this unit means it will not be visible on the Course Overview ' +
+        'page, Section Dialog, or Teacher Dashboard. It will still visible to ' +
+        'Levelbuilders. Would you like to continue with saving?';
```

Comment:
this consolidates 2 warnings which were there previously, and fixes the logic
![Screenshot 2025-05-12 at 3 53 57 PM](https://github.com/user-attachments/assets/73572868-3e5e-485c-9236-14142c950a5a)